--- a/src/types.db
+++ b/src/types.db
@@ -384,3 +384,16 @@ arc_ratio               value:GAUGE:0:U
 arc_size                current:GAUGE:0:U, target:GAUGE:0:U, minlimit:GAUGE:0:U, maxlimit:GAUGE:0:U
 mysql_qcache            hits:COUNTER:0:U, inserts:COUNTER:0:U, not_cached:COUNTER:0:U, lowmem_prunes:COUNTER:0:U, queries_in_cache:GAUGE:0:U
 mysql_threads           running:GAUGE:0:U, connected:GAUGE:0:U, cached:GAUGE:0:U, created:COUNTER:0:U
+
+# qdisc rickk
+qdisc_overlimits	value:COUNTER:0:U
+qdisc_requeues		value:COUNTER:0:U
+qdisc_packets		value:COUNTER:0:U
+
+# dfs-collectd rickk
+phy_errors		value:COUNTER:0:U
+pulse_events		value:COUNTER:0:U
+dfs_pulses_detected	value:COUNTER:0:U
+radars_detected		value:GAUGE:0:U
+channel			value:GAUGE:0:U
+
