--- a/nl80211.h
+++ b/nl80211.h
@@ -2749,8 +2749,6 @@ enum nl80211_commands {
  *	When used with %NL80211_CMD_FRAME_TX_STATUS, indicates the ack RX
  *	timestamp. When used with %NL80211_CMD_FRAME RX notification, indicates
  *	the incoming frame RX timestamp.
- * @NL80211_ATTR_TD_BITMAP: Transition Disable bitmap, for subsequent
- *	(re)associations.
  * @NL80211_ATTR_WIPHY_ANTENNA_GAIN: Configured antenna gain. Used to reduce
  *	transmit power to stay within regulatory limits. u32, dBi.
  *
@@ -3281,7 +3279,6 @@ enum nl80211_attrs {
 
 	NL80211_ATTR_TX_HW_TIMESTAMP,
 	NL80211_ATTR_RX_HW_TIMESTAMP,
-	NL80211_ATTR_TD_BITMAP,
 
 	NL80211_ATTR_WIPHY_ANTENNA_GAIN,
 
